jsonnet/components/control-plane.libsonnet

```
@@ -141,6 +141,61 @@ function(params)
       },
     },
 
+    // This adds a kubelet ServiceMonitor for special use with
+    // prometheus-adapter
+    serviceMonitorKubeletPrometheusAdapter: super.serviceMonitorKubelet + {
```

Using `self` doesn't require to override other fields like `labels`, `selector`, etc.
```suggestion
    serviceMonitorKubeletPrometheusAdapter: self.serviceMonitorKubelet + {
```